oura-library/src/main/kotlin/org/radarbase/oura/request/OuraRequestGenerator.kt

```diff
@@ -94,13 +93,21 @@ constructor(
     }
 
     override fun requestSuccessful(request: RestRequest, response: Response): List<TopicData> {
-        logger.debug("Request successful: {}. Writing to offsets...", request.request)
+        logger.debug("Request successful: {}..", request.request)
         val body: ResponseBody? = response.body
         val data = body?.bytes()!!
         val records = request.route.converters.flatMap {
             it.convert(request, response.headers, data)
         }
-        ouraOffsetManager.updateOffsets(request.route, request.user, request.endDate)
+        val offset = records.maxByOrNull { it -> it.offset }?.offset
+        if (offset != null) {
+            logger.info("Writing ${records.size} records to offsets...")
+            ouraOffsetManager.updateOffsets(
+                request.route,
+                request.user,
+                Instant.ofEpochMilli(offset),
+            )
+        }
```

Comment:
just wondering if offset is null should we throw an exception?

Reply:
@yatharthranjan Ah yes, I was thinking maybe we can just make it request end date if a certain period has passed like we previously discussed? So it will keep retrying until that period is over. I made the changes here: https://github.com/RADAR-base/RADAR-REST-Connector/pull/102/commits/cf65cd135aab09e60cb2033e3ceaaa9da5a7c7a1

What do you think?

Reply:
Maybe we can increase the time? 1 min seems very low, lets discuss tomorrow.